packages/use-dataloader/src/DataLoaderProvider.tsx

```diff
@@ -30,209 +33,186 @@ type UseDataLoaderInitializerArgs<T = unknown> = {
 }
 
 // @ts-expect-error we force the context to undefined, should be corrected with default values
-export const DataLoaderContext = createContext<Context>(undefined)
+export const DataLoaderContext = createContext<IDataLoaderContext>(undefined)
 
 const DataLoaderProvider = ({
   children,
   cacheKeyPrefix,
   onError,
+  maxConcurrentRequests,
 }: {
   children: ReactNode
   cacheKeyPrefix: string
   onError: OnErrorFn
+  maxConcurrentRequests?: number
 }): ReactElement => {
-  const [requestQueue, setRequestQueue] = useState({} as RequestQueue)
-  const [cachedData, setCachedDataPrivate] = useState<CachedData>({})
-  const reloads = useRef<Reloads>({})
+  const [requests, setRequests] = useState<RequestQueue>({})
 
   const computeKey = useCallback(
     (key: string) => `${cacheKeyPrefix ? `${cacheKeyPrefix}-` : ''}${key}`,
     [cacheKeyPrefix],
   )
 
-  const setCachedData = useCallback(
-    (compute: CachedData | ((data: CachedData) => CachedData)) => {
-      if (typeof compute === 'function') {
-        setCachedDataPrivate(current => compute(current))
-      } else {
-        setCachedDataPrivate(compute)
-      }
-    },
-    [],
-  )
-
-  const setReloads = useCallback(
-    (compute: Reloads | ((data: Reloads) => Reloads)) => {
-      if (typeof compute === 'function') {
-        reloads.current = compute(reloads.current)
-      } else {
-        reloads.current = compute
-      }
-    },
-    [],
-  )
-
-  const addCachedData = useCallback(
-    (key: string, newData: unknown) => {
-      if (newData) {
-        if (key && typeof key === 'string') {
-          setCachedData(actualCachedData => ({
-            ...actualCachedData,
-            [computeKey(key)]: newData,
-          }))
-        } else throw new Error(KEY_IS_NOT_STRING_ERROR)
-      }
-    },
-    [setCachedData, computeKey],
-  )
-
-  const addReload = useCallback(
-    (key: string, method: () => Promise<void | unknown>) => {
-      if (method) {
-        if (key && typeof key === 'string') {
-          setReloads(actualReloads => ({
-            ...actualReloads,
-            [computeKey(key)]: method,
-          }))
-        } else throw new Error(KEY_IS_NOT_STRING_ERROR)
-      }
-    },
-    [setReloads, computeKey],
-  )
-
   const addRequest = useCallback(
     (key: string, args: UseDataLoaderInitializerArgs) => {
       if (key && typeof key === 'string') {
         const notifyChanges = (updatedRequest: DataLoader) => {
-          setRequestQueue(current => ({
+          setRequests(current => ({
             ...current,
-            [computeKey(updatedRequest.key)]: updatedRequest,
+            [updatedRequest.key]: updatedRequest,
           }))
         }
-        const newRequest = new DataLoader({ ...args, notify: notifyChanges })
-        newRequest.addOnSuccessListener(result => {
-          if (result !== undefined && result !== null)
-            addCachedData(key, result)
+        const newRequest = new DataLoader({
```

Comment:
`maxConcurrentRequests`

should be passed here rather than set later with `useLayoutEffect`.

Reply:
useLayoutEffect is triggered before the component will be mount in the DOM so it will always set the maxConcurrentRequests before any `useDataLoader` are mounted.

Reply:
I know when it will be triggered ;) it's just weird to set it elsewhere.